### PR TITLE
Upstream play.it.test.ServerEndpoint.SelfSigned

### PR DESCRIPTION
**Pull request by dwijnand**
Also make chunks of FakeKeyStore back into an object.

### CODE REVIEW

File: ssl-config-core/src/main/scala/com/typesafe/sslconfig/ssl/SelfSigned.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * Contains a statically initialized self-signed certificate.
+ *
+ * Was: play.it.test.ServerEndpoint.SelfSigned
+ */
+object SelfSigned {
+  /**
+   * The SSLContext and TrustManager associated with the self-signed certificate.
+   */
+  lazy val (sslContext, trustManager): (SSLContext, X509TrustManager) = {
```
Comment (dwijnand) [inline ssl-config-core/src/main/scala/com/typesafe/sslconfig/ssl/SelfSigned.scala]: Actually, this should be in a dedicated "testkit"-like module (a new one) of ssl-config.

Comment (marcospereira) [reply ssl-config-core/src/main/scala/com/typesafe/sslconfig/ssl/SelfSigned.scala]: Looks like a minor thing to do. Do you want to solve this at this PR or in a new one?

**Comment by dwijnand** [comment]
I'll break this up and fix it.

File: ssl-config-core/src/main/scala/com/typesafe/sslconfig/ssl/SelfSigned.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import java.security.KeyStore
+
+import javax.net.ssl._
+
+/**
+ * Contains a statically initialized self-signed certificate.
+ *
+ * Was: play.it.test.ServerEndpoint.SelfSigned
+ */
+object SelfSigned {
```
Comment (dwijnand) [inline ssl-config-core/src/main/scala/com/typesafe/sslconfig/ssl/SelfSigned.scala]: Call it `FakeSSLTools`

**Pull request closed without merging**